File: eco_buddy/lib/shared/models/auth_response.dart
import 'package:json_annotation/json_annotation.dart';
import 'user.dart';
import '../../features/narration/domain/models/story_history_model.dart';

part 'auth_response.g.dart';

@JsonSerializable()
class AuthResponse {
  final String token;
  final String type;
  @JsonKey(fromJson: _parseExpiresIn)
  final int? expiresIn;      // Durée d'expiration en secondes (Long backend → int)
  final User user;           // Informations utilisateur complètes
  final List<StoryHistoryModel>? storyHistory; // Historique des histoires narratives

  const AuthResponse({
    required this.token,
    this.type = 'Bearer',
    this.expiresIn,
    required this.user,
    this.storyHistory,
  });

  factory AuthResponse.fromJson(Map<String, dynamic> json) => _$AuthResponseFromJson(json);
  Map<String, dynamic> toJson() => _$AuthResponseToJson(this);
  
  // Getters pour compatibilité avec l'ancien code
  String get username => user.username;
  String get email => user.email;
  String get role => user.role;
  int get points => user.points;
}

/// Parse expiresIn from various types (int, String, Long)
int? _parseExpiresIn(dynamic value) {
  if (value == null) return null;
  if (value is int) return value;
  if (value is String) return int.tryParse(value);
  if (value is double) return value.toInt();
  return null;
}

File: ecobuddy-backend/src/main/java/sn/codiallo/ecoBuddy/dto/AuthResponse.java
package sn.codiallo.ecoBuddy.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import sn.codiallo.ecoBuddy.model.User;

import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class AuthResponse {
    // Informations de token
    private String token;
    private String type = "Bearer";
    private Long expiresIn = 86400L; // 24h en secondes
    
    // Informations utilisateur complètes
    private UserResponse user;

    // Historique des histoires narratives
    private List<StoryHistoryResponse> storyHistory;

    public AuthResponse(String token, User user) {
        this.token = token;
        this.user = UserResponse.fromUser(user);
        this.storyHistory = List.of(); // Empty by default
    }

    public AuthResponse(String token, User user, List<StoryHistoryResponse> storyHistory) {
        this.token = token;
        this.user = UserResponse.fromUser(user);
        this.storyHistory = storyHistory != null ? storyHistory : List.of();
    }
    
    // Constructeur legacy pour compatibilité
    public AuthResponse(String token, String username, String email, String role, Integer points) {
        this.token = token;
        this.user = new UserResponse();
        this.user.setUsername(username);
        this.user.setEmail(email);
        this.user.setRole(role);
        this.user.setPoints(points);
    }
}